### PR TITLE
[Polygon Edge] Add FAQ

### PR DESCRIPTION
**Pull request by laviniat1996**
This PR creates an FAQ section for Polygon Edge.

### CODE REVIEW

File: docs/edge/faq/Contracts.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+---
+id: contracts
+title: Smart contracts FAQ
+description: "FAQ for Polygon Edge smart contracts"
```
Comment (zivkovicmilos) [inline docs/edge/faq/Contracts.md:4]: I would capitalize this to `Smart Contracts`

Comment (zivkovicmilos) [reply docs/edge/faq/Contracts.md:4]: Not just here, but everywhere in the doc

File: docs/edge/faq/Contracts.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  - edge
+  - FAQ
+  - smart
+  - contracts
+  
+---
+
+## Does Polygon Edge support smart contracts?
+
+Yes. You can deploy a smart contract in the same manner as Ethereum.
```
Comment (zivkovicmilos) [inline docs/edge/faq/Contracts.md:17]: Why not specifically state that we support ERC20, ERC721...?

Comment (zivkovicmilos) [reply docs/edge/faq/Contracts.md:17]: I would also rephrase this to be:

`Yes. Polygon Edge networks are Ethereum compatible blockchains, so Smart Contracts that are deployable on Ethereum, are also deployable on Polygon Edge chains. `

File: docs/edge/faq/Gas.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+## How to set the gas(native) token total supply?
+
+You can set a premined balance to accounts (addreses) by using the `--premine flag`. Please note that this is a configuration from the genesis file, and it cannot be changed later.
+
+Example on how to use the `--premine flag`:
+
+`--premine=0x3956E90e632AEbBF34DEB49b71c28A83Bc029862:1000000000000000000000`
+
+This sets a premined balance of 1000eth to 0x3956E90e632AEbBF34DEB49b71c28A83Bc029862 (the amount from the argument is in gwei).
```
Comment (zivkovicmilos) [inline docs/edge/faq/Gas.md:29]: The amount is **not** in gwei, it's actually in wei.

Please correct this 🙏

File: docs/edge/faq/Validators.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  - polygon
+  - edge
+  - FAQ
+  - validators
+  
+---
+
+## How to add/remove a validator?
+
+Polygon Edge has PoA and PoS as consensus mechanisms. PoA is the default one. For a new cluster, to switch to PoS, you will need to add the `--pos` flag when generating the genesis file. If you have a running cluster, you can find [here](/docs/edge/consensus/migration-to-pos) how to make the switch. All the info you need about our consensus mechanisms and setup can be found on our [consensus section](/docs/edge/consensus/poa).
```
Comment (zivkovicmilos) [inline docs/edge/faq/Validators.md:16]: How does this answer the question `How do I add a validator`?

File: docs/edge/faq/Validators.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+### PoA
+Adding/removing validators is done by voting. You can find [here](/docs/edge/consensus/poa) a full guide about this.
```
Comment (zivkovicmilos) [inline docs/edge/faq/Validators.md:19]: Typo, should be `You can find a full guide about this here.`

File: docs/edge/faq/Validators.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+### PoS
+You can find [here](/docs/edge/consensus/pos-stake-unstake) a guide on how to stake funds, so that a node can become a validator, and how to unstake (remove the validator).
```
Comment (zivkovicmilos) [inline docs/edge/faq/Validators.md:22]: Typo, should be `You can find a guide on how to stake funds here...`

File: docs/edge/faq/Validators.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+Obs:
```
Comment (zivkovicmilos) [inline docs/edge/faq/Validators.md:24]: What is `0bs`?

File: docs/edge/faq/Validators.md [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+- You can use the genesis flag `--max-validator-count` to set a maximum number of stakers that can join the validator set.
+- You can use the the genesis flag `--min-validator-count ` to set the minimum number of stakers needed to join the validator set(`1` by default).
+- When the maximum validator number is met, you cannot add another validator until you remove an existing one from the set (doesn't matter if the staked amount of the new one is higher). If you remove a validator, the number of validators remaining cannot be lower than `--min-validator-count`.
+- Be aware that there is a default threshold of `1 ETH` for becoming a validator.
```
Comment (zivkovicmilos) [inline docs/edge/faq/Validators.md:28]: It's not `1 ETH`. It's 1 unit of native network (gas) currency.

File: docs/edge/faq/Validators.md [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+ 
+
+
+## How much disk space is recommended for a validator?
+
+We recommend starting with 100G as a conservatively estimated runway, and making sure that it is possible to scale the disk afterwards.
```
Comment (zivkovicmilos) [inline docs/edge/faq/Validators.md:34]: @ZeljkoBenovic

Where did we get this number?

File: sidebars.js [hunk 10]
```diff
@@ -1124,6 +1124,22 @@ module.exports = {
                   ]
               }
           ]
-      }
+      },
+
```
Comment (zivkovicmilos) [inline sidebars.js:1128]: Nitpick: there is an empty space here

File: sidebars.js [hunk 11]
```diff
@@ -1124,6 +1124,22 @@ module.exports = {
...
+        },
+        items: [
+            "edge/faq/validators",
+            "edge/faq/gas",
+            "edge/faq/contracts",
+            "edge/faq/tokens",
+
+        ]
+    }
+
```
Comment (zivkovicmilos) [inline sidebars.js:1143]: Nitpick: there is an empty space here

**Review by zivkovicmilos — CHANGES_REQUESTED** [review]
Looks good 👍 

There are some things that need to be changed, however

File: docs/edge/faq/Gas.md (hunk 3 above)
Comment (laviniat1996) [reply docs/edge/faq/Gas.md]: fixed 3aece441bf8e53b34bcd8c47487e61bb57f1efeb

File: docs/edge/faq/Contracts.md (hunk 1 above)
Comment (laviniat1996) [reply docs/edge/faq/Contracts.md]: fixed daf63233adcbfa43e630fea32a5e945dc5f835c4

File: docs/edge/faq/Contracts.md (hunk 2 above)
Comment (laviniat1996) [reply docs/edge/faq/Contracts.md]: fixed daf63233adcbfa43e630fea32a5e945dc5f835c4

File: docs/edge/faq/Gas.md (hunk 3 above)
Comment (laviniat1996) [reply docs/edge/faq/Gas.md]: Thank you for noticing this.

File: docs/edge/faq/Contracts.md (hunk 2 above)
Comment (laviniat1996) [reply docs/edge/faq/Contracts.md]: Totally agree.

File: docs/edge/faq/Validators.md (hunk 8 above)
Comment (laviniat1996) [reply docs/edge/faq/Validators.md]: Thank you. Fixed the mistake cbea01b269bdb2416a68ad1b7534d2be9e6c9232

File: docs/edge/faq/Validators.md (hunk 7 above)
Comment (laviniat1996) [reply docs/edge/faq/Validators.md]: Obs would be short for observations, but you are right that it is confusing. Fixed cbea01b269bdb2416a68ad1b7534d2be9e6c9232

File: docs/edge/faq/Validators.md (hunk 6 above)
Comment (laviniat1996) [reply docs/edge/faq/Validators.md]: Agree. Fixed cbea01b269bdb2416a68ad1b7534d2be9e6c9232

File: docs/edge/faq/Validators.md (hunk 5 above)
Comment (laviniat1996) [reply docs/edge/faq/Validators.md:17]: Agree. Fixed cbea01b269bdb2416a68ad1b7534d2be9e6c9232

File: docs/edge/faq/Validators.md (hunk 4 above)
Comment (laviniat1996) [reply docs/edge/faq/Validators.md]: I wanted to give some context, but you are right that it's not on point. I've removed that part. Fixed cbea01b269bdb2416a68ad1b7534d2be9e6c9232

File: sidebars.js (hunk 10 above)
Comment (laviniat1996) [reply sidebars.js:1128]: It bothers me too tbh. Fixed b4645794bc3b8f7f5c6b5bdfc27f94094f27e936

File: sidebars.js [hunk 12]
```diff
@@ -1124,6 +1124,22 @@ module.exports = {
...
+        label: "FAQ",
+        link: {
+          type: "generated-index",
+        },
+        items: [
+            "edge/faq/validators",
+            "edge/faq/gas",
+            "edge/faq/contracts",
+            "edge/faq/tokens",
+
```
[the remark this answers is not in the document]
Comment (laviniat1996) [reply sidebars.js]: Fixed b4645794bc3b8f7f5c6b5bdfc27f94094f27e936

File: sidebars.js (hunk 11 above)
Comment (laviniat1996) [reply sidebars.js]: Fixed b4645794bc3b8f7f5c6b5bdfc27f94094f27e936

**Comment by laviniat1996** [comment]
@zivkovicmilos Do you think that this 14b2dee2f28e54907f694195415f2c82ea367e2e addition would be useful?

File: sidebars.js (hunk 10 above)
Comment (zivkovicmilos) [reply sidebars.js:1128]: This empty space is still there 🙂

File: docs/edge/faq/Validators.md [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+- There is a default threshold of `1`  unit of native network(gas) currency for becoming a validator.
+ 
+
+
+## How much disk space is recommended for a validator?
+
+We recommend starting with 100G as a conservatively estimated runway, and making sure that it is possible to scale the disk afterwards.
+
+
+## Is there a maximum number of validators?
```
Comment (zivkovicmilos) [inline docs/edge/faq/Validators.md:35]: I would rephrase this to 
`Is there a limit to the number of validators`?

File: sidebars.js (hunk 10 above)
Comment (laviniat1996) [reply sidebars.js]: :( fixed for good b5cecb2a5b5df07552f12f7f504ef7f0d5a8c9cc

File: docs/edge/faq/Validators.md (hunk 13 above)
Comment (laviniat1996) [reply docs/edge/faq/Validators.md]: Fixed 55cb236ff67bf9e3e1f3bf0e6725a5ad90e845df

File: docs/edge/faq/Validators.md (hunk 9 above)
Comment (zivkovicmilos) [reply docs/edge/faq/Validators.md:32]: Discussed this internally so I'm resolving

**Review by zivkovicmilos — APPROVED** [review]
Thank you for resolving the comments 🙏 

Looks good 💯
